File: haldus-norm/src/main/java/org/featx/sta2ry/haldus/entity/UserSourceEntity.java
package org.featx.sta2ry.haldus.entity;

import lombok.*;
import org.featx.spec.entity.AbstractRecord;

import java.io.Serial;

/**
 * @author excepts
 */
@Data
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserSourceEntity extends AbstractRecord<Long> {
    @Serial
    private static final long serialVersionUID = 4821435260059749785L;

    private String userCode;

    private String service;
}

File: src/main/kotlin/org/featx/sta2ry/haldus/entity/UserSourceEntity.kt
package org.featx.sta2ry.haldus.entity

import org.featx.spec.entity.AbstractRecord

data class UserSourceEntity(
    var service: String? = null,
    var serviceUserCode: String? = null,
    var serviceKey: String? = null,
    var serviceKeyCode: String? = null
) : AbstractRecord<Long>() {
}